apps/web/src/components/milestone-logs/recruitment/MilestoneForm.tsx

```diff
@@ -90,13 +90,19 @@ export const MilestoneForm = <T extends MilestoneFormValues>(props: MilestoneFor
     }
   };
 
-  const handleOutcomeType = (group: string, setFieldValue: FormikHelpers<T>['setFieldValue']) => {
+  const handleOutcomeType = (group: string, { setFieldValue, setFieldTouched }: FormikProps<T>) => {
     const outcomeGroup = Object.values(OutcomeGroups).find(({ value }) => value === group);
 
     if (outcomeGroup?.value === `${STATUS.REFERRAL_ACKNOWLEDGED}`) {
       const milestoneId = milestones.find(s => s.status == outcomeGroup?.value);
       setFieldValue('status', milestoneId?.id);
+    } else {
+      setFieldValue('status', '');
+      setFieldTouched('status', false);
```

Comment:
I still get the issue.  But I think the underlying issue is the Milestone doesn't get reset on form submit, we just need to add a check for the value in the BasicSelect.tsx.  something like `value={value ? options.find(o => o.value === value) : null}`

Reply:
I didn't mean to reset by submitting, but I'll do that too.